src/autoqasm/simulator/program_context.py

```diff
@@ -0,0 +1,224 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from functools import singledispatchmethod
+from typing import Optional, Union
+
+import numpy as np
+from braket.default_simulator.openqasm._helpers.arrays import (
+    convert_discrete_set_to_list,
+    convert_range_def_to_slice,
+)
+from braket.default_simulator.openqasm.circuit import Circuit
+from braket.default_simulator.openqasm.parser.openqasm_ast import (
+    ClassicalType,
+    DiscreteSet,
+    Identifier,
+    IndexedIdentifier,
+    IntegerLiteral,
+    RangeDefinition,
+    SymbolLiteral,
+)
+from braket.default_simulator.openqasm.program_context import ProgramContext, Table
+from braket.default_simulator.operation import GateOperation
+from sympy import Integer
+
+from autoqasm.simulator.conversion import convert_to_output
+
+
+class QubitTable(Table):
```

Comment:
Any reason not to just add this to the main branch of the default simulator's `QubitTable`?

Reply:
No, I don't see any reason why we couldn't do that. I have opened issue https://github.com/amazon-braket/autoqasm/issues/20 to track this.